chunk-test.py

```diff
@@ -0,0 +1,91 @@
+import os
+import random
+import shutil
+import tempfile
+from typing import BinaryIO, Dict, Iterable
+import uuid
+
+# Input data from the client
+
+chunks = [
+    dict(fileName="fileA", chunkNumber=1, chunkContents=b"chunkA1\n",),
+    dict(fileName="fileA", chunkNumber=2, chunkContents=b"chunkA2\n",),
+    dict(fileName="fileB", chunkNumber=1, chunkContents=b"chunkB1\n",),
+    dict(fileName="fileB", chunkNumber=2, chunkContents=b"chunkB2\n",),
+    dict(fileName="fileB", chunkNumber=3, chunkContents=b"chunkB3\n",),
+]
+random.shuffle(chunks)
+
+chunked_upload_id = str(uuid.uuid4())
+
+
+# Overall approach:
+# For each file we want to upload:
+# 1. Upload each chunk to a separate file
+# 2. When we're done uploading all the chunks, concatenate the chunks into a single file
+# 3. Open the concatenated file as a file stream
+#
+# Note that this allows the chunks to be uploaded in any order (or even in parallel).
+
+
+def chunked_upload_dir_path(chunked_upload_id: str):
+    tempdir_path = tempfile.gettempdir()
+    return os.path.join(tempdir_path, chunked_upload_id)
+
+
+# Handle one chunk upload (this would be one API endpoint)
+def upload_chunk(chunked_upload_id: str, chunk):
+    # Store each chunk in a separate file
+    # /tmp/<chunked_upload_id>/<file_name>/<chunk_number>
+    file_dir_path = os.path.join(
+        chunked_upload_dir_path(chunked_upload_id), chunk["fileName"]
+    )
+    os.makedirs(file_dir_path, exist_ok=True)
+    chunk_path = os.path.join(file_dir_path, str(chunk["chunkNumber"]))
+    print("Uploading", chunk_path)
+    with open(chunk_path, "wb") as chunk_file:
+        chunk_file.write(chunk["chunkContents"])
```

Comment:
How big do we think these contents are going to be? Could/should we stream this rather than loading the whole thing in memory?

Reply:
I'm still trying to figure out what a good chunk size is. Maybe somewhere around 1-5MB? Streaming it should be pretty easy if needed